File: src/main/java/com/github/resources/AdminResource.java
package com.github.resources;

import com.github.repository.AdminRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

/**
 * Created by zk_chs on 16/4/11.
 */
@RestController
@RequestMapping(value = "/admin")
public class AdminResource {

    @Autowired
    private AdminRepository adminRepository;

    @RequestMapping(value = "/{id}", method = RequestMethod.GET)
    public Object get (@PathVariable(value = "id") int id){
        return adminRepository.findOne(id);
    }

}

File: src/main/scala/com/github/resources/AdminResource.scala
package com.github.resources

import com.github.repository.AdminRepository
import com.github.utils.M
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.{RequestParam, RequestMethod, RequestMapping, RestController}

/**
  * Created by zk_chs on 16/4/13.
  */
@RestController
@RequestMapping(value = Array("/admin"))
class AdminResource @Autowired() (adminRepository: AdminRepository) {

  @RequestMapping(method = Array(RequestMethod GET, RequestMethod POST))
  def get(@RequestParam(value = "id", required = false) id: Int) {
    if ( id == 0 ) M builder() put ("status", "fail") put ("msg", "id may not be null") buildMap()
    else M builder() put("status", "success") put("result", adminRepository findOne id) buildMap()
  }

}
